networkmanager/networkmanager_test.go

```diff
@@ -71,542 +120,924 @@ func TestMain(m *testing.M) {
+		t.Fatalf("Should be an error: traffic monitoring is disabled")
+	}
 
-	if ulSpeed > float64(setULSpeed)*delta || dlSpeed > float64(setDLSpeed)*delta {
-		t.Errorf("Speed limit exceeds expected level: DL %0.2f, UL %0.2f", dlSpeed, ulSpeed)
+	if err := manager.SetTrafficPeriod(0); err == nil {
+		t.Fatalf("Should be an error: traffic monitoring is disabled")
 	}
 }
 
-/*******************************************************************************
+/***********************************************************************************************************************
  * Private
- ******************************************************************************/
+ **********************************************************************************************************************/
 
-func setup() (err error) {
-	if tmpDir, err = ioutil.TempDir("", "aos_"); err != nil {
-		return aoserrors.Wrap(err)
+func (storage *testTrafficStorage) SetTrafficMonitorData(chain string, timestamp time.Time, value uint64) error {
+	if storage.disableSaveTraffic {
+		return aoserrors.New("Problem to save traffic")
 	}
 
-	if err = os.MkdirAll(tmpDir, 0755); err != nil {
-		return aoserrors.Wrap(err)
+	storage.chains[chain] = trafficData{lastUpdate: timestamp, currentValue: value}
+
+	return nil
+}
+
+func (storage *testTrafficStorage) GetTrafficMonitorData(chain string) (timestamp time.Time, value uint64, err error) {
+	if storage.disableLoadTraffic {
+		return timestamp, 0, aoserrors.New("Problem to load traffic")
 	}
 
-	if manager, err = networkmanager.New(&config.Config{WorkingDir: tmpDir}, nil); err != nil {
-		return aoserrors.Wrap(err)
+	data, ok := storage.chains[chain]
+	if !ok {
+		return timestamp, 0, networkmanager.ErrNotExist
 	}
 
+	return data.lastUpdate, data.currentValue, nil
+}
+
+func (storage *testTrafficStorage) RemoveTrafficMonitorData(chain string) error {
+	if _, ok := storage.chains[chain]; !ok {
+		return networkmanager.ErrNotExist
+	}
+
+	delete(storage.chains, chain)
+
 	return nil
 }
 
-func cleanup() {
-	if err := manager.DeleteAllNetworks(); err != nil {
-		log.Errorf("Can't remove networks: %s", err)
+func createPlugins(plugins []string) string {
+	networkConfig := `{"name":"network0","cniVersion":"0.4.0","plugins":[`
+
+	for i, plugin := range plugins {
+		networkConfig += plugin
+		if i != len(plugins)-1 {
+			networkConfig += `,`
+		}
 	}
 
-	manager.Close()
+	return networkConfig + `]}`
+}
 
-	if err := os.RemoveAll(tmpDir); err != nil {
-		log.Errorf("Can't remove tmp dir: %s", err)
+func readFromFile(path string) (content string, err error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return "", aoserrors.Wrap(err)
 	}
+
+	defer func() {
+		if err = file.Close(); err != nil {
+			log.Error(err)
+		}
+	}()
+
+	b, err := ioutil.ReadAll(file)
+	if err != nil {
+		return "", aoserrors.Wrap(err)
+	}
+
+	content = string(b)
+
+	content = strings.ReplaceAll(content, " ", "")
```

Comment:
strings package does not contain such a TrimSpaces function, there is TrimSpace, according to the documentation which removes all leading and trailing white space, in the current case, need to remove all spaces in the string

Reply:
Add new function `removeSpaces`. It can be implemented like this:
```go
func removeSpaces(str string) string {
    return strings.Map(func(r rune) rune {
        if unicode.IsSpace(r) {
	    return -1
	}
        
        return r
    }, str)
}
```